gostatsd.go

```diff
@@ -22,36 +25,126 @@ var (
 	Version string
 )
 
-func main() {
-	runtime.GOMAXPROCS(runtime.NumCPU())
-	s := statsd.NewServer()
-	s.AddFlags(pflag.CommandLine)
-	pflag.Parse()
+const (
+	ParamVerbose    = "verbose"
+	ParamCPUProfile = "cpu-profile"
+	ParamJson       = "json"
+	ParamConfigPath = "config-path"
+	ParamVersion    = "version"
+)
 
-	if s.CPUProfile != "" {
-		f, err := os.Create(s.CPUProfile)
+const EnvPrefix = "GSD" //Go Stats D
+
+func main() {
+	err, v, version := setConfiguration()
+	if err != nil {
+		log.Fatalf("Error while parsing configuration: %v\n", err)
+	}
+	if version {
```

Comment:
Tnagent: we should extract version to its own package. I didn't do it previously because I was struggling to set it in the `Makefile` when not in `main` but I know how to do it properly now.